frontend/components/canvas.tsx

```diff
@@ -1,75 +1,85 @@
-"use client";
+'use client'
 
-import { useDraw } from "@/hooks/useDraw";
-import { useState } from "react";
-import { CompactPicker } from "react-color";
+import { FC, useEffect, useState } from 'react'
+import { useDraw } from '../hooks/useDraw'
+import { ChromePicker } from 'react-color'
 
-export default function Canvas() {
-  const [color, setColor] = useState<string>('#000');
-  const [width, setWidth] = useState<number>(5);
-  const { canvasRef, onMouseDown, clear } = useDraw(drawLine);
+import { io } from 'socket.io-client'
+import { drawLine } from '../util/drawLine'
+const socket = io('http://localhost:3001')
```

Comment:
Instead of creating a new socket instance, just update the Canvas component to accept a socket instance and use that. 

See how the `Chat` component in `frontend\components\chat.tsx` is accepting a socket 

and how `frontend\app\rooms\[roomId]\page.tsx` initializes that Chat component

Reply:
we initialize the socket connection once in `frontend\app\rooms\[roomId]\page.tsx` and I would like to avoid initializing a second connection by creating a new instance here

Reply:
yea this was a placeholder since I was still figuring out how to merget into the socket you wrote